### DISCUSSION TITLE
Contributing

### DISCUSSION THREAD

**Discussion by blaize** [fyne-io/fyne#4042] [Q&A] [answered]
Hello,
I made 2 news cool features to fyne develop branch. How to share it ? I checked the Pull Request thing but I'm not sure I understood very well how it work.

Thanks.

**Comment by Jacalz** [comment, marked as answer]
That sounds intriguing. Feel free to share more information :)

The easiest way to see how to create a PR is likely to follow some guides online. I think it would be hard to write it all down here. Some examples:

- https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request

- https://www.digitalocean.com/community/tutorials/how-to-create-a-pull-request-on-github


What are those features? Before you open a PR with a lot of work it can be good for use to know what it is we can see how it fits in to our plan and vision for the project. I.e. some work might find a better place in [fyne-x](https://github.com/fyne-io/fyne-x) (we also land new widgets and such there sometimes to mature a bit before bringing it into regular Fyne), some might not fit into our vision of the toolkit and so on.

**Reply by blaize** [reply]
Hello,
Nothing crazy. I made a patch to set an importance to a label. This make the label change the color depending of the importance.
The other feature is about themed icons, I found it missing WarningThemedResource and SuccessThemedResource to display an icon with the corresponding color.
I needed thoses feature for my app, so I made it. The UI is much more intuitive with colors. Maybe it can be usefull for other people.
Do you think it worth ?

Thanks for the link, I will read it.

**Reply by Jacalz** [reply]
For the first one, it doesn't hurt to open a PR. I can't say anything about if it will be accepted or not but you can always give it a shot :)

The second one seems to now have some discussion on the issue in question. It might be worth waiting

**Reply by blaize** [reply]
Thanks you. Very excited to learn how to do it. I'm not a pro coder but I hope I can help in any way to improve the toolkit ! Even if it's just as user feedback.

**Reply by Jacalz** [reply]
Any help is appreciated. Welcome onboard
